### ISSUE TITLE
wazuh-db error during syscollector deltas handling

### ISSUE THREAD

**Issue by jnasselle** [wazuh/wazuh#14987]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| v4.4 | wazuh-db | Manager | Any | Linux |

## Description

It was found that wazuh-db is having some errors during syscollector deltas coming from a Windows Server 2022

```
wazuh-db: ERROR: (5215): DB(011) SQL Error: 'not an error'.
```
After setting wazuh-db debug level to 2
```
2022/09/20 14:23:55 wazuh-db[8298] wdb_parser.c:257 at wdb_parse(): DEBUG: Agent 001 query: dbsync network_iface MODIFIED 2022/09/20 14:23:44|Loopback Pseudo-Interface 1|Software Loopback Interface 1| |NULL|4294967295.000000|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|NULL|
2022/09/20 14:23:55 wazuh-db[8298] wdb_delta_event.c:172 at wdb_modify_dbsync(): ERROR: (5215): DB(001) SQL Error: 'not an error'.
```
Agent's `network_iface` inventory 
```
0|2022/09/20 13:47:34|Loopback Pseudo-Interface 1|Software Loopback Interface 1| |up||00:00:00:00:00:00|0|0|0|0|0|0|0|0|e4a92faeea2613dab29a772ec7f3540be951b478|56e07d1bb3aeca5febd3a26616defa115f6a881b
```

## DoD

- [ ] RCA
- [ ] Fix this issue
- [ ] Improve error handling (SQL Error: 'not an error')
- [ ] Add propper UT
- [ ] QA automatized test 
- [ ] QA manual test

**Comment by jnasselle** [comment]
# RCA

- Fists impressions:

`wdb_dbsync_stmt_bind_from_string` return errors are always treated as SQLITE errors, but this is not true. 

This function could fail if a casting failure happens, meaning that the stated type and the item type are different. For example, if the field has `FIELD_INTEGER` type but the field is string/float or whatever type that does not allow `strtol` to work.

This error message handling should be fixed. Unit tests and Integrations test should be added too.

- Deeper analysis

Despite errors being unproperly handled, we need to find those values that are rising this errors and figure out if they are caused by the agent and the delta information or is located on the syscollector delta decoder

**Comment by jnasselle** [comment]
This issue will be addressed in #15639 fix
